### PR TITLE
feat: add __newArrayBuffer() to the loader

### PR DESCRIPTION
**Pull request by FGasper**
- [x] I've read the contributing guidelines

### CODE REVIEW

**Comment by FGasper** [comment]
Nudging @MaxGraey and @dcodeIO (and any other maintainers?) … am I just waiting at this point for someone to approve the workflow run for this PR?

Thank you!

**Review by MaxGraey — APPROVED** [review]

File: lib/loader/tests/index.js [hunk 1]
```diff
@@ -36,6 +36,17 @@ function test(file) {
...
 
+  // should be able to allocate and work with a new small ArrayBuffer
+  {
+    let u8arr = new Uint8Array([1, 2, 3, 4]);
+    let buf = u8arr.buffer;
+    let ref = exports.__newArrayBuffer(buf);
+    let got = exports.__getArrayBuffer(ref);
+    let gotu8arr = new Uint8Array(got);
+
+    assert.strictEqual(gotu8arr.join(), u8arr.join());
```
Comment (MaxGraey) [inline lib/loader/tests/index.js:47]:
```suggestion
    assert.deepStrictEqual(gotu8arr, u8arr);
```

**Comment by MaxGraey** [comment]
It seems your email doesn't associate with github account?

**Comment by FGasper** [comment]
> It seems your email doesn't associate with github account?

I have it set to private. I’ll update the PR.

File: lib/loader/tests/index.js [hunk 2]
```diff
@@ -36,6 +36,17 @@ function test(file) {
...
 
+  // should be able to allocate and work with a new small ArrayBuffer
+  {
+    let u8arr = new Uint8Array([1, 2, 3, 4]);
+    let buf = u8arr.buffer;
+    let ref = exports.__newArrayBuffer(buf);
+    let got = exports.__getArrayBuffer(ref);
+    let gotu8arr = new Uint8Array(got);
+
+    assert.deepStrictEqual(gotu8arr, u8arr);
```
Comment (MaxGraey) [inline lib/loader/tests/index.js:47]:
```suggestion
    let input  = new Uint8Array([1, 2, 3, 4]);
    let bufPtr = exports.__newArrayBuffer(input.buffer);
    let output = new Uint8Array(exports.__getArrayBuffer(bufPtr));

    assert.deepStrictEqual(output, input);
```

File: NOTICE [hunk 3]
```diff
@@ -41,6 +41,7 @@ under the licensing terms detailed in LICENSE:
 * Ryan Pivovar <ryanpivovar@gmail.com>
 * Roman F. <70765447+romdotdog@users.noreply.github.com>
 * Joe Pea <trusktr@gmail.com>
+* Felipe Gasper <fgasper@users.noreply.github.com>
```
Comment (MaxGraey) [inline NOTICE:44]:
```suggestion
* Felipe Gasper <FGasper@users.noreply.github.com>
```

**Review by MaxGraey — APPROVED** [review]

**Review by dcodeIO — APPROVED** [review]
👍

**Comment by MaxGraey** [comment]
Thanks!

**Pull request merged**